### PR TITLE
Shardcontroller enhancements

### PR DESCRIPTION
**Pull request by rtay1188**
Add root parameter values and delete flow logging

### CODE REVIEW

File: admiral/cmd/admiral/cmd/root.go [hunk 1]
```diff
@@ -246,6 +246,10 @@ func GetRootCmd(args []string) *cobra.Command {
 	rootCmd.PersistentFlags().BoolVar(&params.AdmiralOperatorMode, "admiral_operator_mode", false, "Enable/Disable admiral operator functionality")
 	rootCmd.PersistentFlags().StringVar(&params.OperatorSyncNamespace, "operator_sync_namespace", "admiral-operator-sync",
 		"Namespace in which Admiral Operator will put its generated configurations")
+	rootCmd.PersistentFlags().StringVar(&params.LabelSet.OperatorIdentityLabel, "operator_identity_label", "admiral.io/operatorIdentity", "used to filter which shard Admiral Operator will watch")
```
Comment (nirvanagit) [inline admiral/cmd/admiral/cmd/root.go:249]: OperatorIdentityLabelKey

File: admiral/cmd/admiral/cmd/root.go [hunk 2]
```diff
@@ -246,6 +246,10 @@ func GetRootCmd(args []string) *cobra.Command {
...
+	rootCmd.PersistentFlags().StringVar(&params.LabelSet.ShardIdentityLabel, "shard_identity_label", "admiral.io/shardIdentity", "used to filter which shard Admiral Operator will watch")
```
Comment (nirvanagit) [inline admiral/cmd/admiral/cmd/root.go:250]: ShardIdentityLabelKey

File: admiral/pkg/controller/admiral/shard.go [hunk 3]
```diff
@@ -149,9 +149,10 @@ func NewShardController(stopCh <-chan struct{}, handler ShardHandler, configPath
 	if err != nil {
 		return nil, fmt.Errorf("failed to create shard controller crd client: %v", err)
 	}
-	//TODO: should not be hardcoded, fetch actual expected operator and shard identities from env variables
 	//labelOptions := informers.WithTweakListOptions(func(opts *metav1.ListOptions) {
-	//	opts.LabelSelector = "admiral.io/operatorIdentity=operatorIdentity, admiral.io/shardIdentity=dev"
+	//  opIdLabel, opIdValue := common.GetOperatorIdentityLabelValueSet()
+	//  shardIdLabel, shardIdValue := common.GetShardIdentityLabelValueSet()
+	//  opts.LabelSelector = fmt.Sprintf("%s=%s, %s=%s", opIdLabel, opIdValue, shardIdLabel, shardIdValue)
```
Comment (nirvanagit) [inline admiral/pkg/controller/admiral/shard.go:155]: Why do we need dead code?

File: admiral/pkg/controller/common/config.go [hunk 4]
```diff
@@ -471,6 +471,18 @@ func GetOperatorSyncNamespace() string {
 	return wrapper.params.OperatorSyncNamespace
 }
 
+func GetOperatorIdentityLabelValueSet() (string, string) {
```
Comment (nirvanagit) [inline admiral/pkg/controller/common/config.go:474]:
```suggestion
func GetOperatorIdentityLabelKeyValueSet() (string, string) {
```

File: admiral/pkg/controller/common/config.go [hunk 5]
```diff
@@ -471,6 +471,18 @@ func GetOperatorSyncNamespace() string {
...
+	wrapper.RLock()
+	defer wrapper.RUnlock()
+	return wrapper.params.LabelSet.OperatorIdentityLabel, wrapper.params.OperatorIdentityValue
+}
+
+func GetShardIdentityLabelValueSet() (string, string) {
```
Comment (nirvanagit) [inline admiral/pkg/controller/common/config.go:480]:
```suggestion
func GetShardIdentityLabelKeyValueSet() (string, string) {
```

File: admiral/pkg/controller/common/types.go [hunk 6]
```diff
@@ -154,6 +157,8 @@ type LabelSet struct {
 	GatewayApp                          string //the value for `app` key that will be used to fetch the loadblancer for cross cluster calls, also referred to as east west gateway
 	AdmiralCRDIdentityLabel             string //Label Used to identify identity label for crd
 	IdentityPartitionKey                string //Label used for partitioning assets with same identity into groups
+	OperatorIdentityLabel               string
```
Comment (nirvanagit) [inline admiral/pkg/controller/common/types.go:160]:
```suggestion
	OperatorIdentityLabelKey              string
```

File: admiral/pkg/controller/common/types.go [hunk 7]
```diff
@@ -154,6 +157,8 @@ type LabelSet struct {
...
+	ShardIdentityLabel                  string
```
Comment (nirvanagit) [inline admiral/pkg/controller/common/types.go:161]:
```suggestion
	ShardIdentityLabelKey                  string
```

File: admiral/pkg/controller/admiral/shard.go [hunk 8]
```diff
@@ -149,12 +149,12 @@ func NewShardController(stopCh <-chan struct{}, handler ShardHandler, configPath
...
 		return nil, fmt.Errorf("failed to create shard controller crd client: %v", err)
 	}
-	//TODO: should not be hardcoded, fetch actual expected operator and shard identities from env variables
-	//labelOptions := informers.WithTweakListOptions(func(opts *metav1.ListOptions) {
-	//	opts.LabelSelector = "admiral.io/operatorIdentity=operatorIdentity, admiral.io/shardIdentity=dev"
-	//})
-	//informerFactory := informers.NewSharedInformerFactoryWithOptions(shardController.K8sClient, resyncPeriod, labelOptions)
-	informerFactory := informers.NewSharedInformerFactoryWithOptions(shardController.K8sClient, resyncPeriod)
+	labelOptions := informers.WithTweakListOptions(func(opts *metav1.ListOptions) {
+		opIdLabel, opIdValue := common.GetOperatorIdentityLabelKeyValueSet()
```
Comment (nirvanagit) [inline admiral/pkg/controller/admiral/shard.go:153]: Can we use a constant label key, and provide ability to configure different values. That way we will standardize the key amongst all admiral operators.

File: admiral/pkg/clusters/shard_handler.go [hunk 9]
```diff
@@ -35,11 +35,12 @@ func (sh *ShardHandler) Added(ctx context.Context, obj *admiralapiv1.Shard) erro
 }
 
 func (sh *ShardHandler) Deleted(ctx context.Context, obj *admiralapiv1.Shard) error {
-	// TODO: Not yet implemented
 	//err := HandleEventForShard(ctx, admiral.Delete, obj, sh.RemoteRegistry)
```
Comment (vinay-g) [inline admiral/pkg/clusters/shard_handler.go:38]: was this intentional ?

Comment (rtay1188) [reply admiral/pkg/clusters/shard_handler.go:38]: Yes, there is nothing to do in handler for delete - the controller removes from cache, and we don't want operator to try to recreate the shard or anything like that

**Review by vinay-g — APPROVED** [review]

**Pull request merged**